### ISSUE TITLE
Add underscore methods `_.findIndex, `_.findLastIndex` and  `_.sortedIndex`

### ISSUE THREAD

**Issue by jacobbuck** [jashkenas/backbone#3809]
Finding myself writing:

```js
var index = collection.indexOf(collection.findWhere({foo: 'bar'}));
```

when this is so much better:

```js
var index = collection.findIndex({foo: 'bar'});
```

**Comment by jridgewell** [comment]
Want to work on a PR?

**Comment by jacobbuck** [comment]
Sure, just submitted one now. Though I'm not sure how to approach adding `_.sortedIndex` as it's iteratee and value arguments are in reverse order to the other underscore methods.

**Comment by jridgewell** [comment]
Closed by #3811.
